library/blob.c

```diff
@@ -491,6 +695,23 @@ static int eblob_check_disk_one(struct eblob_iterate_local *loc)
 		}
 	}
 
+	if ((ctl->flags & EBLOB_ITERATE_FLAGS_VERIFY_CHECKSUM) &&
+	    !(dc->flags & BLOB_DISK_CTL_REMOVE) &&
+	    !(dc->flags & BLOB_DISK_CTL_UNCOMMITTED)) {
+		struct eblob_write_control wc;
+		memset(&wc, 0, sizeof(wc));
+		eblob_rctl_to_wc(&rc, &wc);
+		eblob_dc_to_wc(dc, &wc);
+
+		err = eblob_verify_checksum(bc->back, &dc->key, &wc);
+		if (err) {
+			eblob_dump_wc(bc->back, &dc->key, &wc, "eblob_check_disk_one: checksum verification failed", err);
+			eblob_mark_entry_removed_purge(bc->back, &dc->key, &rc);
```

Comment:
Recovery didn't remove those keys yet. And I do hope it will not.
This change will immediately remove such keys in my setup, I will not apply this pull request.

Also, you didn't add this similar and even stronger feature (when headers heavily mismatch) into eblob read call, and now want to put this very subtle version into iterator. If there is a chance of removing valid data, it can not go in.

Ideal case would be iterator to fix such keys - if headers in data and index file are mismatched, iterator removes those keys, if there is no footer (or it doesn't have enough space), iterator sets no-checksum bit, if index and data sizes mismatch, index is set to what is present in data and so on.